### ISSUE TITLE
DPI awareness

### ISSUE THREAD

**Issue by ProgerXP** [ProgerXP/Notepad2e#154]
Recent laptops often have high-DPI displays and Windows enables DPI scaling by default. Unlike scaling in XP, scaling in Vista+ (?) applies not to font alone but to all controls. When a program isn't "dpi-aware", Windows resorts to proportional scaling of the entire window which blurs the text. For a text editor, results are awful. 

Question: how hard it is to make this program "dpi-aware" so that text and controls are scaled properly and nothing is blurred?

Screenshot of DPI setting in Display settings:
![dpi1](https://user-images.githubusercontent.com/1643195/36505164-8db012fa-174a-11e8-8b9a-041df8db5fc7.png)

Screenshot of Notepad 2e with DPI at 125% (note the difference between font used in the main menu and inside the program's window itself):
![dpi2](https://user-images.githubusercontent.com/1643195/36505165-8dda61ea-174a-11e8-9b88-9d919c12a67c.png)

(To see the difference clearer open the image in a new tab.)

**Comment by cshnik** [comment]
Please check uploaded test build with implemented DPI awareness (per monitor) functionality.
Since this feature affects many aspects of the application it would be great to have a detailed test before committing these changes. Here is the list of known limitations in this build (found for Windows 10):
1. System dialogs are not scaled (when moving them between monitors).
2. Tool/setting windows captions (e.g. Settings> Tab Settings) are not scaled.
3. There can be some resource leaks, additional code cleanup is required.

**Comment by ProgerXP** [comment]
It seems to work fine. Push your code changes to the repo.

**Comment by cshnik** [comment]
Feature is broken in [R134](https://github.com/ProgerXP/Notepad2e/releases/tag/R134):
Scintilla's content is not properly scaled.

**Comment by ProgerXP** [comment]
Is it another problem caused by the Scintilla upgrade?

**Comment by cshnik** [comment]
> Is it another problem caused by the Scintilla upgrade?

Yes, there was a missing change when upgrading Scintilla. Fixed.

There is another issue: high-DPI scaling is not applied for system dialogs (i.e when running MessageBox()).
Please confirm if we should address it.

**Comment by ProgerXP** [comment]
> There is another issue: high-DPI scaling is not applied for system dialogs (i.e when running MessageBox()).
 Please confirm if we should address it.

Do you have ideas for how to address it?

**Comment by cshnik** [comment]
Fixed bug: DPI scaling is not applied on application start which caused UI looks smaller (than expected) on high-DPI displays.
Workaround: move Notepad2e to another display and then return it back.

**Comment by ProgerXP** [comment]
>  Move Notepad2e to Display 1 and then return it back to Display 2.

Is this workaround for the version before c8218bc83b138c262d530dbdbc1e59188a9c1113 only?

---

It was suggested to fix DPI of system windows by subclassing; this will also make the code cleaner. Let's do this and see if it will work.
